### ISSUE TITLE
BUG: Selection of custom component type not working in the canvas

### ISSUE THREAD

**Issue by DreamCloudProject** [GrapesJS/grapesjs#5334]
### GrapesJS version

- [X] I confirm to use the latest version of GrapesJS

### What browser are you using?

Chrome, Opera, Edge, etc

### Reproducible demo link

https://jsfiddle.net/crszh0jo/6/

### Describe the bug

I'm trying to make it work the same case as it described here [[QUESTION] Selection of custom components not possible from canvas #951](https://github.com/GrapesJS/grapesjs/issues/951), but no luck.

How can I make selection from canvas work the same way as selection from layer manager? Because the solution described here does not work for my case. I need to allow the user to select a custom component from the canvas instead of opening the layer manager every time.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by DreamCloudProject** [comment]
https://github.com/GrapesJS/grapesjs/assets/7961936/12180b41-4c29-489d-9bcd-1ea025a96a8f

This example shows how it works in my editor.

**Comment by yatoogamii** [comment]
Hi @DreamCloudProject , it seems like your component Msc-header is locked. it's an attribute which allow the component to be clicked from the layers but not from the canva itself

Can you share or verify your component definition and check if the "locked" attributes is set to true ?

**Comment by artf** [comment]
There are a few caveats to deal with when you're using custom components with a shadow host.
By default, the host has the display as `inline` and causes issues with styling.

You can decide to force the display on the component level itself or update the view of the GrapesJS component, eg.
```js
view: {
  onAttrUpdate() {
    this.el.style.display = 'block';
  }
}
```

**Comment by DreamCloudProject** [comment]
Thanks, that partially worked and solved another problem! Could you please reopen this thread because I still can't select msc-header. I can't reproduce this problem externally, because I have LitElement used as the main framework to run grapejs.

**Comment by DreamCloudProject** [comment]
> Hi @DreamCloudProject , it seems like your component Msc-header is locked. it's an attribute which allow the component to be clicked from the layers but not from the canva itself
> 
> Can you share or verify your component definition and check if the "locked" attributes is set to true ?

I have not found any clues that could indicate this, could you please explain in detail where and what I should check?

**Comment by DreamCloudProject** [comment]
https://github.com/GrapesJS/grapesjs/assets/7961936/af2c230e-ab60-44fe-9e4e-03984b015c40

I can only provide a demo instance where you can test this issue directly. The above video shows how to test this issue.

https://demo.mastermindcms.co

Login: demo
Password: hello

**Comment by yatoogamii** [comment]
I'm not familiar with LitElement so it will be hard to explain more in detail. In vanilla grapesjs when creating a custom component type you could provide different properties as draggable, badgable, locked and so on. example:

```ts
  editor.Components.addType("ctm-row", {
    model: {
      defaults: {
        name: "Row",
        tagName: "div",
        draggable: (target: grapesjs.Component, dest: grapesjs.Component) => {
          return dest.attributes.type !== "ctm-row";
        },
       locked: true // Disable the selection of the component and its children in the canvas. Default: false
    },
  });
```

Your component behavior seems similar to this, but it might not be related if its a bug describe above or if LitElement create your components for you already.

I didn't find any clue either by seeing the demo you provide. Sorry !

**Comment by DreamCloudProject** [comment]
> I'm not familiar with LitElement so it will be hard to explain more in detail. In vanilla grapesjs when creating a custom component type you could provide different properties as draggable, badgable, locked and so on. example:
> 
> ```ts
>   editor.Components.addType("ctm-row", {
>     model: {
>       defaults: {
>         name: "Row",
>         tagName: "div",
>         draggable: (target: grapesjs.Component, dest: grapesjs.Component) => {
>           return dest.attributes.type !== "ctm-row";
>         },
>        locked: true // Disable the selection of the component and its children in the canvas. Default: false
>     },
>   });
> ```
> 
> Your component behavior seems similar to this, but it might not be related if its a bug describe above or if LitElement create your components for you already.
> 
> I didn't find any clue either by seeing the demo you provide. Sorry !

I noticed that if you remove the 100% width inside shadow host for the header tag, it can be selected.

https://github.com/GrapesJS/grapesjs/assets/7961936/01ff79c4-b5c1-4fc5-be92-9ea93d4ca90b

**Comment by artf** [comment]
@DreamCloudProject not sure without the code, but it looks like you're stopping the propagation of clicks somewhere in the component

**Comment by DreamCloudProject** [comment]
@artf Yes in some components in ShadowDOM I do that. Because there are situations when a parent component reacts to a click, and its children have to handle the click on their own events.

But it's still possible to select a component, as the layer manager does. Why not use such a function on the canvas?

**Comment by artf** [comment]
> Why not use such a function on the canvas?

@DreamCloudProject Well... maybe because clicking inside the canvas and the layers is not the same thing 😅. 
How do you think the canvas catches component selection/clicks?! There is one listener on top of the document, if you stop the click propagation, no click is happened for the editor.

**Comment by DreamCloudProject** [comment]
@artf Yes, you're right. I'm blocking the click propagation, and so the editor doesn't receive the click event, hence can't execute the logic to select the element.

I tried setting the click propagation only on the current element by replacing `e.stopPropagation()` with `e.preventDefault()` inside the component, and now the element selection logic in the editor works. The problem seems to be solved.
